### DISCUSSION TITLE
auto command TabLeave

### DISCUSSION THREAD

**Discussion by Isos9** [neovim/neovim#29933] [Q&A] [answered]
Hi,

I have a keymap that I add in a specific filetype that opens in a new tab, which is simply "q" to call ":tabclose", and I added an autocmd to delete this keymap once the tab is closed, but it didn't work anymore after I added the autocmd TabLeave, because some of my actions can take me back to the prev tab and I don't need the other one to stay open, does anyone know why it stopped working with this new autocmd ? I don't understand.

Here is the code:

```lua
vim.keymap.set('n', 'q', '<cmd>tabclose<CR>', { noremap = true })

vim.api.nvim_create_autocmd('TabClosed', {
	callback = function()
		vim.keymap.del('n', 'q')
	end,
	once = true,
})

vim.api.nvim_create_autocmd('TabLeave', {
	command = ':tabclose',
	once = true,
})
```

Thanks

**Comment by zeertzjq** [comment, marked as answer]
You need to add `nested = true` to the `TabLeave` autocommand.

**Reply by Isos9** [reply]
Ok thank you, that works
